packages/promotion/src/services/promotion-module.ts

```diff
@@ -71,6 +76,162 @@ export default class PromotionModuleService<
     return joinerConfig
   }
 
+  async computeActions(
+    promotionsToApply: Pick<PromotionTypes.PromotionDTO, "code">[],
```

Comment:
```suggestion
    promotionsToApply: Required<Pick<PromotionTypes.PromotionDTO, "code">>[],
```

Reply:
choosing the array of strings over this.